GCTC-NTGC/gc-digital-talent
# Feature - 3661 - Application Snapshot

Resolves #3661 

## Summary

This triggers the profile snapshot laid out in #3707 

## What This Does

Adds a new event `ApplicationSubmitted` that is dispatched in the `submitApplication` mutation. `StoreApplicationSnapshot` listens to the event and stores the profile snapshot when it is dispatched.

## Testing

1. Start by clearing app cache `php artisan event:clear`
2. Ensure the event exists and has the listener `php artisan event:list`
3. Make a `PoolCandidate` that is ready to be submitted
  - Complete profile
  - All required skills
  - `DRAFT` status
  - `null` for the following
    - `submitted_at`
    - `signature`
    - `profile_snapshot`
4. Run the `submitApplication` mutation
5. Ensure the snapshot was saved with the proper `JSON`
